src/MeetingsSDKAdapter.js

```diff
@@ -509,6 +590,157 @@ export default class MeetingsSDKAdapter extends MeetingsAdapter {
+      sdkMeeting.emit(EVENT_MEDIA_LOCAL_UPDATE, {
+        control: SHARE_CONTROL,
+        state: MeetingControlState.INACTIVE,
+      });
+
+      await sdkMeeting.updateShare({
+        sendShare: false,
+        receiveShare: true,
+      });
+
+      this.stopStream(this.meetings[ID].localShare);
+      this.meetings[ID].localShare = null;
+    };
+
+    const resetSharingStream = () => {
+      // eslint-disable-next-line no-console
+      console.error(`Unable to update local share for meeting "${ID}"`);
+      sdkMeeting.emit(EVENT_MEDIA_LOCAL_UPDATE, {
+        control: SHARE_CONTROL,
+        state: MeetingControlState.INACTIVE,
+      });
+
+      if (this.meetings[ID].localShare) {
+        // eslint-disable-next-line no-console
+        console.warn(`Stop local share stream for meeting "${ID}"`);
+        this.stopStream(this.meetings[ID].localShare);
+        this.meetings[ID].localShare = null;
+      }
+    };
+
+    //
+    // Workflow:
+    // To enable or to disable the local sharing stream based on toggle state.
+    // Will stop sharing stream and reset UI state when error happens
+    //
+    try {
+      if (this.meetings[ID].localShare) {
+        await disableSharingStream();
+      } else {
+        await enableSharingStream();
+      }
+    } catch (error) {
+      resetSharingStream();
+    }
+  }
+
+  /**
+   * Returns an observable that emits the display data of a share control.
+   *
+   * @param {string} ID ID of the meeting to start/stop share
+   * @returns {Observable.<MeetingControlDisplay>}
+   * @memberof MeetingJSONAdapter
+   */
+  shareControl(ID) {
+    const sdkMeeting = this.fetchMeeting(ID);
+    const shareInactiveMsg = {
+      ID: SHARE_CONTROL,
+      icon: 'share',
+      tooltip: 'Start Share',
+      state: MeetingControlState.INACTIVE,
+      text: null,
+    };
+    const shareActiveMsg = {
+      ID: SHARE_CONTROL,
+      icon: 'share',
+      tooltip: 'Stop Share',
+      state: MeetingControlState.ACTIVE,
+      text: null,
+    };
+    const shareBusyMsg = {
+      ID: SHARE_CONTROL,
+      icon: 'share',
+      tooltip: 'Share is busy',
+      state: MeetingControlState.DISABLED,
+      text: null,
+    };
+
+    const getDisplayData$ = Observable.create((observer) => {
+      if (sdkMeeting) {
+        observer.next(shareInactiveMsg);
+      } else {
+        observer.error(new Error(`Could not find meeting with ID "${ID}" to add share control`));
+      }
+
+      observer.complete();
+    });
+
+    const localMediaUpdateEvent$ = fromEvent(sdkMeeting, EVENT_MEDIA_LOCAL_UPDATE).pipe(
+      filter((event) => event.control === SHARE_CONTROL),
+      map(({state}) => {
+        switch (state) {
+          case MeetingControlState.DISABLED:
+            return shareBusyMsg;
+          case MeetingControlState.INACTIVE:
+            return shareInactiveMsg;
+          case MeetingControlState.ACTIVE:
+            return shareActiveMsg;
+          default:
+            return null;
```

Comment:
I think the control should be disabled by default (e.g. going into some unknown state)

Reply:
So you think it is better to return the disabledShare object for default?

Reply:
Yes because if for whatever reason the code ends up in the `default` case, that means something we didn't expect emitted the event and we potentially won't know how to handle the flow. Therefore we should allow the user to use the share control since we won't know what could happen (even potentially crash).